tests/test_web_sendfile_functional.py

```diff
@@ -1164,3 +1164,30 @@ async def handler(request: web.Request) -> web.FileResponse:
 
     resp.release()
     await client.close()
+
+
+@pytest.mark.parametrize("chunk_size", (15, 31))
+async def test_a_ranged_request_serves_no_additional_bytes(
+    aiohttp_client: AiohttpClient, sender: _Sender, chunk_size: int
+) -> None:
+    filepath = pathlib.Path(__file__).parent / "aiohttp.jpg"
+
+    async def handler(request: web.Request) -> web.FileResponse:
+        return sender(filepath, chunk_size=chunk_size)
+
+    app = web.Application()
+    app.router.add_get("/", handler)
+    client = await aiohttp_client(app)
+
+    resp = await client.get("/", headers={"Range": "bytes=20-40"})
+    assert resp.status == 206
+    body = await resp.read()
```

Comment:
It may also be important to use the context manager correctly if we want t test 2 requests in a row.
```suggestion
    async with client.get("/", headers={"Range": "bytes=20-40"}) as resp:
        assert resp.status == 206
        body = await resp.read()
```